pkgs/servers/nfd/default.nix

```diff
@@ -16,34 +17,34 @@
 
 stdenv.mkDerivation rec {
   pname = "nfd";
-  version = "22.12";
+  version = "24.07";
 
   src = fetchFromGitHub {
     owner = "named-data";
     repo = lib.toUpper pname;
     rev = "NFD-${version}";
-    hash = "sha256-epY5qtET7rsKL3KIKvxfa+wF+AGZbYs+zRhy8SnIffk=";
+    hash = "sha256-iEI8iS0eLLVe6PkOiCHL3onYNVYVZ1ttmk/aWrBkDhg=";
     fetchSubmodules = true;
   };
 
   nativeBuildInputs = [ pkg-config sphinx wafHook ];
-  buildInputs = [ libpcap ndn-cxx openssl websocketpp ] ++ lib.optional withSystemd systemd;
+  buildInputs = [ dpkg libpcap ndn-cxx openssl websocketpp ] ++ lib.optional withSystemd systemd;
 
   wafConfigureFlags = [
-    "--boost-includes=${boost179.dev}/include"
-    "--boost-libs=${boost179.out}/lib"
-    "--with-tests"
+    "--boost-includes=${boost.dev}/include"
+    "--boost-libs=${boost.out}/lib"
+    # "--with-tests" # fatal error: boost/test/unit_test.hpp: No such file or directory
```

Comment:
If tests are failing, I do wonder whether it's wise to update to that version instead of waiting for upstream fixes to land?

Reply:
Unfortunately devs managing the upstream are known to do releases once in a while and their tests are half broken. This has been the case for the last few years, so I don't have much faith in them fixing them. I tried to keep as many tests as possible that would work, but now they don't even compile correctly.